test/utils.test.ts

```diff
@@ -117,10 +123,238 @@ describe('utils', function () {
+        nonce: 7,
+        chainId: 33,
+        data: 'Dummy Deploy Data',
+        gasLimit: constants.Zero,
+        value: constants.Zero,
+        to: undefined,
+        from: relayWorkerAddress,
+      };
+
+      expect(() =>
+        validateRelayResponse(
+          deployRequest,
+          transaction,
+          relayWorkerAddress
+        )
+      ).to.throw(`Transaction has no recipient address`);
+    });
+
+    it('Should throw error if transaction has no sender address', function () {
+      const transaction: Transaction = {
+        nonce: 7,
+        chainId: 33,
+        data: 'Dummy Deploy Data',
+        gasLimit: constants.Zero,
+        value: constants.Zero,
+        to: relayHubAddress,
+        from: undefined,
+      };
+
+      expect(() =>
+        validateRelayResponse(
+          deployRequest,
+          transaction,
+          relayWorkerAddress
+        )
+      ).to.throw(`Transaction has no signer`);
+    });
+
+    it('Should throw error if nonce is greater than relayMaxNonce', function () {
+      const transaction: Transaction = {
+        nonce: 7,
+        chainId: 33,
+        data: 'Dummy Deploy Data',
+        gasLimit: constants.Zero,
+        value: constants.Zero,
+        to: relayHubAddress,
+        from: relayWorkerAddress,
+      };
+
+      expect(() =>
+        validateRelayResponse(
+          deployRequest,
+          transaction,
+          relayWorkerAddress
+        )
+      ).to.throw(
+        `Relay used a tx nonce higher than requested. Requested ${deployRequest.metadata.relayMaxNonce.toString()} got ${transaction.nonce}`
+      );
+    });
+
+    it('Should throw error if Transaction recipient is not the RelayHubAddress', function () {
+      const transaction: Transaction = {
+        nonce: 5,
+        chainId: 33,
+        data: 'Dummy Deploy Data',
+        gasLimit: constants.Zero,
+        value: constants.Zero,
+        to: 'Dummy Address',
+        from: relayWorkerAddress,
+      };
+
+      expect(() =>
+        validateRelayResponse(
+          deployRequest,
+          transaction,
+          relayWorkerAddress
+        )
+      ).to.throw(`Transaction recipient must be the RelayHubAddress`);
+    });
+
+    it('Should throw error if Relay request Encoded data is not the same as Transaction data', function () {
+      const transaction: Transaction = {
+        nonce: 5,
+        chainId: 33,
+        data: 'Dummy Different Data',
+        gasLimit: constants.Zero,
+        value: constants.Zero,
+        to: relayHubAddress,
+        from: relayWorkerAddress,
+      };
+
+      expect(() =>
+        validateRelayResponse(
+          deployRequest,
+          transaction,
+          relayWorkerAddress
+        )
+      ).to.throw(
+        `Relay request Encoded data must be the same as Transaction data`
+      );
+    });
+
+    it('Should throw error if Relay Worker is not the same as Transaction sender', function () {
+      // FIXME: fix test
```

Comment:
Is this comment still valid?

Reply:
Im not pretty sure, @ironFe93, do you know if its still valid? Im also checking by myself.

Reply:
I just check it and for me, its valid, it has a different worker address

Reply:
so, in order to fix the test, should we change the worker address being used?

Reply:
but the test itself needs to throw an error what is currently doing, if I put the worker that is being used, it will not throw the error

Reply:
sure, got it!